File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/discovery/ServiceDiscoveryRequestSpec.kt
package stasis.test.client_android.lib.discovery

import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe
import stasis.client_android.lib.discovery.ServiceDiscoveryRequest

class ServiceDiscoveryRequestSpec : WordSpec({
    "A ServiceDiscoveryRequest" should {
        "support converting its attributes to a request ID" {
            val request = ServiceDiscoveryRequest(
                isInitialRequest = false,
                attributes = mapOf(
                    "b" to "42",
                    "c" to "false",
                    "a" to "test-string"
                )
            )

            request.id shouldBe ("a=test-string::b=42::c=false")
        }
    }
})

File: core/src/test/scala/stasis/test/specs/unit/core/discovery/ServiceDiscoveryRequestSpec.scala
package stasis.test.specs.unit.core.discovery

import stasis.core.discovery.ServiceDiscoveryRequest
import io.github.sndnv.layers.testing.UnitSpec

class ServiceDiscoveryRequestSpec extends UnitSpec {
  "A ServiceDiscoveryRequest" should "support converting its attributes to a request ID" in {
    val request = ServiceDiscoveryRequest(
      isInitialRequest = false,
      attributes = Map(
        "b" -> "42",
        "c" -> "false",
        "a" -> "test-string"
      )
    )

    request.id should be("a=test-string::b=42::c=false")
  }
}
